### PR TITLE
Faster Real Time Maps

### PR DESCRIPTION
**Pull request by sgiehl**
Improves fetching visits for realtime map as mentioned in https://github.com/matomo-org/matomo/issues/12722#issuecomment-383105741

refs #12722

### CODE REVIEW

File: plugins/UserCountryMap/javascripts/realtime-map.js [hunk 1]
```diff
@@ -67,28 +67,23 @@
...
                 map = self.map,
-                main = $('.RealTimeMap_container', this.$element),
-                worldTotalVisits = 0,
                 maxVisits = config.maxVisits || 100,
                 changeVisitAlpha = typeof config.changeVisitAlpha === 'undefined' ? true : config.changeVisitAlpha,
                 removeOldVisits = typeof config.removeOldVisits === 'undefined' ? true : config.removeOldVisits,
                 doNotRefreshVisits = typeof config.doNotRefreshVisits === 'undefined' ? false : config.doNotRefreshVisits,
                 enableAnimation = typeof config.enableAnimation === 'undefined' ? true : config.enableAnimation,
                 forceNowValue = typeof config.forceNowValue === 'undefined' ? false : +config.forceNowValue,
-                width = main.width(),
```
[the remark this answers is not in the document]
Comment (sgiehl) [reply plugins/UserCountryMap/javascripts/realtime-map.js]: didn't see any usage of that variable, or did I miss something?

**Comment by tsteur** [comment]
In my case it resulted in this query:
```sql
SELECT sub.* FROM ( SELECT log_visit.* FROM piwik_log_visit AS log_visit WHERE log_visit.idsite in (?) AND log_visit.visit_last_action_time >= ? ORDER BY idsite DESC, visit_last_action_time DESC LIMIT 0, 100 ) AS sub GROUP BY sub.idvisit ORDER BY sub.visit_last_action_time DESC LIMIT 100
```
with bind 
```php
array ( 0 => '1', 1 => '2018-03-25 02:23:01', 
```

This looks good but wondering if we should reduce the last 30 days it looks at by default, change it to say `last3` which should be still good enough for a real time map considering we show like only visits from the last 24 hours or so.

**Comment by sgiehl** [comment]
That should depend on the currently chosen date any period. If you are looking at today, it should only query for todays data. But as it's a realtime map guess we could change that to lower ranges in another PR.
We could also improve the way the query is built. If a minTimestamp is given the query will still contain two checks for time, one with the current period/date and one for the given minTimestamp...

**Comment by tsteur** [comment]
The date selector was disabled in real time map when I tested it and the last30 comes from the controller.

**Pull request merged**